### DISCUSSION TITLE
Support for linear solver Alternating Anderson–Richardson method (AAR)

### DISCUSSION THREAD

**Discussion by FlyingCat2023** [ginkgo-project/ginkgo#1556] [Ideas]
Hi Developers,

I am very interested in the solver developed by Phanish Suryanarayana, published in the paper entitled "Alternating Anderson–Richardson method - An efficient alternative to preconditioned Krylov methods for large, sparse linear systems". 
The authors concluded that this solver is as robust as GMRES, and as efficient as BiCGS. I wonder if there any plan to enable this solver in Ginkgo?

**Comment by pratikvn** [comment]
Unfortunately, we are stretched a bit thin. If you would like to implement it, we would be happy to guide you through the process and help you getting it merged into Ginkgo.

**Reply by FlyingCat2023** [reply]
Hello, I am sure if I can implement the solver by myself. However, if you can provide me a guidance, I will try it later on.
